server/src/routes/v2/apps.js

```diff
@@ -60,8 +70,140 @@ module.exports = [
+            plugins: {
+                'hapi-swagger': {
+                    payloadType: 'form',
+                },
+            },
+        },
+        handler: async (request, h) => {
+            if (!canCreateApp(request, h)) {
+                throw Boom.unauthorized()
+            }
+
+            const { db } = h.context
+            const { id: currentUserId } = await getCurrentUserFromRequest(
+                request,
+                db
+            )
+            const isManager = currentUserIsManager(request)
+
+            const { payload } = request
+            const appJsonPayload = JSON.parse(payload.app)
+            const appJsonValidationResult = CreateAppModel.def.validate(
+                appJsonPayload
+            )
+
+            if (appJsonValidationResult.error) {
+                throw Boom.badRequest(appJsonValidationResult.error)
+            }
+
+            const { organisationId } = appJsonPayload.developer
+            const organisation = await Organisation.findOne(
+                organisationId,
+                false,
+                db
+            )
+            if (!organisation) {
+                throw Boom.badRequest('Unknown organisation')
+            }
+
+            const isMember = await Organisation.hasUser(
+                organisationId,
+                currentUserId,
+                db
+            )
+            if (!isMember && !isManager) {
+                throw Boom.unauthorized(
+                    `You don't have permission to upload apps to that organisation`
+                )
+            }
+
+            const app = await db.transaction(async trx => {
+                const { appType } = appJsonPayload
+                const app = await App.create(
+                    {
+                        userId: currentUserId,
+                        organisationId,
+                        appType,
+                        status: AppStatus.PENDING,
+                    },
+                    trx
+                )
+
+                const { name, description, sourceUrl } = appJsonPayload
+                const {
+                    version,
+                    demoUrl,
+                    minDhisVersion,
+                    maxDhisVersion,
+                    channel,
+                } = appJsonPayload.version
+                const appVersion = await App.createVersionForApp(
+                    app.id,
+                    {
+                        userId: currentUserId,
+                        version,
+                        demoUrl,
+                        sourceUrl,
+                        minDhisVersion,
+                        maxDhisVersion: maxDhisVersion || '',
+                        channel,
+                        appName: name,
+                        description: description || '',
+                    },
+                    trx
+                )
+
+                const { logo } = payload
+                const logoMetadata = logo.hapi
```

Comment:
Lets integrate the newly merged mimeType-validation : https://github.com/dhis2/app-hub/pull/461
Should be just a matter of 
`validateImageMetadata(request.server.mime, logoMetadata`